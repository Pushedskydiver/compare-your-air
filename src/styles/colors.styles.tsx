export function rgba(hex: string, opacity: number) {
  const hexCode = hex.replace('#', '');
  const r = parseInt(hexCode.substring(0, 2), 16);
  const g = parseInt(hexCode.substring(2, 4), 16);
  const b = parseInt(hexCode.substring(4, 6), 16);
  const result = `rgba(${r},${g},${b},${opacity})`;

  return result;
}

export const colors = {
  core: {
    alpha: '#7430a2',
    beta: '#376aa2'
  },
  light: '#ffffff',
  dark: '#222222',
  neutral: '#bebebe',
  pale: '#f2f2f2'
};
