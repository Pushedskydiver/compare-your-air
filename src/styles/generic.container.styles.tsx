import styled from 'styled-components';
import spacing from './spacing.styles';

export const Wrapper = styled.section`
  position: relative;
  max-width: 1200px;
  margin: 0 auto;
`;

export const Container = styled(Wrapper)`
  padding-top:  ${spacing.padding * 4}px;
  padding-right:  ${spacing.padding * 2}px;
  padding-left: ${spacing.padding * 2}px;
`;
