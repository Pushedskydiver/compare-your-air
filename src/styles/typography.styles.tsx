function calcFontSize(number: number) {
  return `${number / 16}em`;
}

export const typography = {
  family: 'Open Sans, sans-serif',
  lineHeight: {
    heading: 1.3,
    body: 1.7
  },
  heading: {
    size: {
      tiny: `calc(${calcFontSize(12)} + 0.5vw)`,
      small: `calc(${calcFontSize(14)} + 0.5vw)`,
      regular: `calc(${calcFontSize(16)} + 0.5vw)`,
      medium: `calc(${calcFontSize(20)} + 0.5vw)`,
      big: `calc(${calcFontSize(30)} + 0.5vw)`,
      large: `calc(${calcFontSize(34)} + 0.5vw)`,
      xl: `calc(${calcFontSize(40)} + 0.5vw)`
    }
  },
  body: {
    size: {
      tiny: `calc(${calcFontSize(11)} + 0.1vw)`,
      small: `calc(${calcFontSize(13)} + 0.1vw)`,
      regular: `calc(${calcFontSize(16)} + 0.1vw)`,
      medium: `calc(${calcFontSize(18)} + 0.1vw)`,
      large: `calc(${calcFontSize(20)} + 0.1vw)`
    }
  },
  weight: {
    regular: '400',
    medium: '600'
  }
};
