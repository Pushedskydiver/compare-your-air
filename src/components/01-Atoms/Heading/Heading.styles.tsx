import styled, { css } from 'styled-components';
import { typography } from '../../../styles/typography.styles';
import spacing from '../../../styles/spacing.styles';

interface HeadingProps {
  color?: string,
  align?: string,
  noMargin?: boolean,
  tiny?: boolean,
  small?: boolean,
  regular?: boolean,
  medium?: boolean,
  big?: boolean,
  large?: boolean,
  xl?: boolean
}

export const Heading1 = styled.h1<HeadingProps>`
  font-family: var(--font);
  font-weight: ${typography.weight.medium};
  line-height: ${typography.lineHeight.heading};

  ${props => props.color && css`
    color: ${props.color};
  `}

  ${props => props.align && css `
    text-align: ${props.align};
  `}

  ${props => !props.noMargin && css`
    margin: 0 0 ${(spacing.margin * 2)}px 0;
  `}

  ${props => props.noMargin && css`
    margin: 0;
  `}

  ${props => props.tiny && css`
    font-size: ${typography.heading.size.tiny};
  `}

  ${props => props.small && css`
    font-size: ${typography.heading.size.small};
  `}

  ${props => props.regular && css`
    font-size: ${typography.heading.size.regular};
  `}

  ${props => props.medium && css`
    font-size: ${typography.heading.size.medium};
  `}

  ${props => props.big && css`
    font-size: ${typography.heading.size.big};
  `}

  ${props => props.large && css`
    font-size: ${typography.heading.size.large};
  `}

  ${props => props.xl && css`
    font-size: ${typography.heading.size.xl};
  `}
`;

export const Heading2 = Heading1.withComponent('h2');
export const Heading3 = Heading1.withComponent('h3');
export const Heading4 = Heading1.withComponent('h4');
export const Heading5 = Heading1.withComponent('h5');
