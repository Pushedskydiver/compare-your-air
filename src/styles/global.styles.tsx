import { createGlobalStyle } from 'styled-components';
import { colors } from './colors.styles';
import { typography } from './typography.styles';
import spacing from './spacing.styles';

const GlobalStyles = createGlobalStyle`
  :root {
    --background-alpha: ${colors.core.alpha};
    --background-beta: ${colors.core.beta};
    --foreground-dark: ${colors.dark};
    --foreground-light: ${colors.light};
    --font: ${typography.family};
  }

  *,
  *:before,
  *:after {
    box-sizing: border-box;
    color: inherit;
    font-weight: normal;
  }

  html,
  body {
    position: relative;
    min-height: 100%;
  }

  html {
    font-size: 16px;
    -ms-text-size-adjust: 100%;
    -webkit-text-size-adjust: 100%;
  }

  body {
    margin: 0;
    background: linear-gradient(135deg, var(--background-alpha) 0%,var(--background-beta) 100%);
    color: var(--foreground-light);
    font-family: var(--font);
    font-weight: ${typography.weight.regular};
    line-height: ${typography.lineHeight.body};
    text-rendering: optimizeLegibility;
    -webkit-font-smoothing: antialiased;
  }

  body ::-moz-selection {
    background: var(--foreground-dark);
    color: var(--foreground-light);
  }

  body ::selection {
    background: var(--foreground-dark);
    color: var(--foreground-light);
  }

  h1 {
    font-family: inherit;
    font-size: ${typography.heading.size.big};
    font-weight: 600;
    line-height: 1.3;
    margin: 0px 0px 18px;
  }

  h1,
  h2 {
    margin-top: 0;
    margin-bottom: ${spacing.margin}px;
  }

  p {
    margin-top: 0;
    margin-bottom: 0;
  }

  strong {
    font-weight: bold;
  }

  input {
    font-family: inherit;
    appearance: none;
  }
`;

export default GlobalStyles;
