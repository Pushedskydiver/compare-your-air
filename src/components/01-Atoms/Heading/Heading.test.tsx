import React from 'react';
import { cleanup, render } from '@testing-library/react';;
import { Heading1 } from './Heading.styles';

afterEach(() => {
  cleanup();
});

describe('<HeadingStyles />', () => {
  it('should match snapshot', () => {
    const { container } = render(
      <Heading1>This is a h1 tag</Heading1>
    );

    expect(container).toMatchSnapshot();
  })
})