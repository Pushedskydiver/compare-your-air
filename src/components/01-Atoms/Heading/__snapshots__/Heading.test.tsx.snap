// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeadingStyles /> should match snapshot 1`] = `
<div>
  <h1
    class="sc-AxjAm cJVohz"
  >
    This is a h1 tag
  </h1>
</div>
`;
